crates/bindings-csharp/BSATN.Runtime.Tests/Tests.cs

```diff
@@ -0,0 +1,179 @@
+namespace SpacetimeDB;
+
+using CsCheck;
+using Xunit;
+
+public static class BSATNRuntimeTests
+{
+    [Fact]
+    public static void AddressRoundtrips()
+    {
+        var str = "00112233445566778899AABBCCDDEEFF";
+        var addr = Address.FromHexString(str);
+
+        Assert.NotNull(addr);
+        Assert.Equal(addr.ToString(), str);
+
+        byte[] bytes =
+        [
+            0x00,
+            0x11,
+            0x22,
+            0x33,
+            0x44,
+            0x55,
+            0x66,
+            0x77,
+            0x88,
+            0x99,
+            0xaa,
+            0xbb,
+            0xcc,
+            0xdd,
+            0xee,
+            0xff,
+        ];
+        var addr2 = Address.FromBigEndian(bytes);
+        Assert.Equal(addr2, addr);
+
+        Array.Reverse(bytes);
+        var addr3 = Address.From(bytes);
+        Assert.Equal(addr3, addr);
+
+        var memoryStream = new MemoryStream();
+        var writer = new BinaryWriter(memoryStream);
+        var bsatn = new Address.BSATN();
+        if (addr is { } addrNotNull)
+        {
+            bsatn.Write(writer, addrNotNull);
+        }
+        else
+        {
+            Assert.Fail("Impossible");
+        }
+        writer.Flush();
```

Comment:
Very minor nitpick, but I'd instead wrap this into a block and use `using ` for memorystream&writer inside, so that RAII takes care of flushing for us.